tools/jenkins/tests.groovy

```diff
@@ -176,5 +185,56 @@ for (def x in slaves) {
 timeout(240) {
     timestamps {
         parallel builders
+
+        if (env.ENABLE_SONAR && currentBuild.result == 'SUCCESS') {
+            node('SLAVE') {
+                stage('SonarQube Analysis') {
+                    try {
+                        checkout_custom()
+
+                        def jdk = tool name: 'java-8-oracle'
+                        env.JAVA_HOME = "${jdk}"
+                        def mvnHome = tool name: 'maven-3.3', type: 'hudson.tasks.Maven$MavenInstallation'
+                        
+                        dir('sources') {
+                            for (def slave in slaves) {
+                                try {
+                                    unstash "coverage_${slave}"
+                                    sh "mv .coverage .coverage.${slave}"
+                                    echo "Unstashed .coverage.${slave}"
+                                } catch(e) {
+                                    currentBuild.result = 'UNSTABLE'
+                                    echo e
+                                }
```

Comment:
In case of error, you can keep `throw e` because the rest of the job will have bad results.